src/environment.ts

```diff
@@ -80,6 +82,17 @@ export class Environment {
     get debugMode(): boolean {
         return this.variables.debugMode;
     }
+
+    get isAllowedLevel(): boolean {
+        const allowedLevelsPaths = ['project', 'vendor'];
```

Comment:
There are a lot of usages of magic string `project` and `vendor`. Please move into configuration as constants.